### ISSUE TITLE
What are the plans for Hubot?

### ISSUE THREAD

**Issue by joeyguerra** [hubotio/hubot#1541]
Hi. I was just wondering if there are any plans for continuing Hubot or not?

**Comment by Eeemil** [comment]
My guess is "no", can any Hubot dev confirm? @mistydemeo

**Comment by xurizaemon** [comment]
I emailed this to opensource@github.com, the contact email displayed at https://github.com/hubotio, May 6 2021. I will update if I hear back - I should have linked to this issue.
----
Hi there

A commonly used Hubot package (hubot-redis-brain) has recently started
failing npm audit checks due to a dependency that needs updating.

There hasn't been any update from the Hubot team in a long time, and I
believe Github's focus is now on the Probot / Botkit interfaces. There
is still a community of Hubot usage though, and I'm interested if you
have any plans you can share for how this community might proceed?

Would Github consider handing the reins over? I don't want ownership
myself, but I do want to see Github support a process to move towards
community ownership if that's plausible.

Thanks

Chris
----
cc @technicalpickles @mistydemeo - are you able to give *any* guidance / contact which would support some kind of community organisation here?

**Comment by mistydemeo** [comment]
I'm afraid I no longer work at GitHub, so while I'm still a Hubot maintainer I'm not able to drive the conversation from that perspective.

**Comment by xurizaemon** [comment]
Thanks for the reply! It might be fine to have any active maintainer, if there are ways we can support you in the issue queue etc. Is the project still on your radar sufficiently that this would be helpful? How could some kind of community support work best for you (and/or other @hubotio maintainers)?

**Comment by dominik-ba** [comment]
Is there another possibility then forking this repo in order to keep this alive?
I e.g. see lots of hello world issues by users already deleted and PRs to bump versions etc. I guess if some of the users could get the permissions to at least triage issues this could get a few updates. Or is there a predecessor which should be used instead of hubot?

**Comment by xurizaemon** [comment]
@dominik-ba I'm interested in ensuring Hubot gets at least the maintenance to continue operation. Simply unblocking version bump PRs would be a step in the right direction. Like you I'd prefer to avoid forking the project if there's some way to build on what exists.

**Comment by jglovier** [comment]
Howdy folks! It's really cool to see so much community interest still in Hubot. I used to use Hubot a lot as well, and I worked on the docs site a bit many years ago, and I still think of Hubot as one of the simplest ways to spin up a basic bot for chatops.

Also FWIW, as a former GitHub staffer I'm assuming that it is not likely for GitHub to release maintainership of Hubot anytime in the immediate future, if for no other reason then because there are still probably a number of apps and deployment processes running at GH on this repo (speculating a bit here, so I could be wrong about this).

So it may be worth seriously considering the fork option if you'd like to drive Hubot into the future (could even call it hubot-future or hubot-next. 😉 ).

The other option, of course, is to get on Twitter and get lots of folks to ask new(ish) GitHub CEO [Thomas Dohmke](https://twitter.com/ashtom) to release this project to the community! (That used to work pretty well with Nat Friedman, I hear 😄). Again, my hunch is this approach is probably a long shot, but still might be worth a try.

**Comment by jglovier** [comment]
There: I [tweeted with a request for a response](https://twitter.com/jglovier/status/1522682924180541441). Feel free to add your comments to the thread, or tweet Thomas in your own thread!

**Comment by xurizaemon** [comment]
I think you're right, and it's fine for Github not to pass on stewardship of Hubot - that's a big ask tbh.

Feels like there's room for a community fork here, a nice thing about which is a fork doesn't borrow from some parts (PRs) and not others (what's with all the hello-world issues?). & in [hubotio/evolution](https://github.com/hubotio/evolution) the project even comes with a (school-age, free to ignore) roadmap, which is a rare feature in unmaintained projects 😆 

My brain parses `hubot-future` (hubot-dash-thing) as "a Hubot plugin that provides Future functionality" which is going to get confusing. nubot? yubot? Are we picking a name here?

**Comment by joeyguerra** [comment]
Forking sounds like the right next step. I'm new here, so a bit clueless on how to create a fork (not git clone, but the social aspect of it) of an open sourced project and bring the community along. Any recommendations?

**Comment by scarolan** [comment]
I love Hubot and have been using it since the Hipchat days. Later when the world moved to Slack I brought the Hubots along.  Slack has written a kind of tutorial to migrate Hubot apps to Bolt, but it seems nobody's published a complete example of a Slack bot running Bolt for JavaScript with clear examples.

Since Hubot seems to have been abandoned...anyone have a good standalone example of a basic Bolt app that has all the main features of Hubot?

**Comment by joeyguerra** [comment]
Just to clarify, I'm still using Hubot. I forked it and have been updating it.

**Comment by joh6nn** [comment]
@joeyguerra i'm Just Some Guy, so take my suggestions with several grains of salt, but I'd say next steps would be:

1. rename your fork so that
  a. it's obvious that development has moved elsewhere
  b. you can create a new organization for it and not have the name conflict with this org
2. reach out to all the people listed as devs on this repo and invite them to contribute to your new fork
3. reach out to anyone who as committed to their own fork in the last ...18 months? and invite them to submit PRs to your fork
4. announce the fork on various platforms (eg HN, maybe some Reddit communities)

I'm on board to help out with this kind of administrative work (i have next to 0 modern JS experience, so that's all I can contribute)

**Comment by joeyguerra** [comment]
@joh6nn this plan is actually quite helpful. Thank you.

**Comment by xurizaemon** [comment]
@joeyguerra if you're keen to take it forward - I'd be much more confident if the changes on your fork had clear intent, eg by taking the changes as they are and reworking them into MRs for each feature or bug fix applied.

If it's time to create a new org, we can do that by lifting and reviewing the changes into that new location. You've clearly done lots of work, and this process would document that.

I realize this is extra labor which you don't owe anyone, but it might really help viability of the fork going forward, which I'm enthusiastic about. Are you up for that?

**Comment by joeyguerra** [comment]
Yes. I think that’s the right approach as well. To clarify:

1. new org
2. Fresh fork into new org
3. MRs for to collaborate on changes

Any ideas on the new org name? Naming is hard.

**Comment by joeyguerra** [comment]
Btw, #hubot on the fediverse.

**Comment by xurizaemon** [comment]
Let's just call the new org "hubot-new" or something, and make an issue for naming over there. Github organisations can be renamed (right?), so we can defer that and deal with it later IMO.

That also gives us something for the new roadmap.

**Comment by joeyguerra** [comment]
Ok. I intend to create an org called hubot-new, add you to it, and fork the repo today.

**Comment by joeyguerra** [comment]
Here's the new org/repo https://github.com/hubot-new/hubot-b

**Comment by xurizaemon** [comment]
@levenleven since you recently merged to this project after a long period of no updates or response from Github, I am pinging you into this discussion to make sure you're aware of plans to explore continuing Hubot support in the new organisation linked above.

I think Hubot users would love to see Github's engagement, and we're also prepared to put some work in ourselves. If Github has plans, now is a great time to clarify them :)

**Comment by joeyguerra** [comment]
Update: In touch with @technicalpickles to review, merge #1581, and create a new release.

**Comment by technicalpickles** [comment]
@joeyguerra reached out over email, and we've been discussing what to do there w/ @xurizaemon and @joh6nn 

They have been working over in https://github.com/hubot-new/hubot to merge PRs, update dependencies and etc. After reviewing, I didn't see any reason to not just add them as maintainers, and get that work merged here before it diverges.

Since then, @joeyguerra started https://github.com/hubotio/hubot/pull/1581 with all the changes, and we've tested, and I just merged it earlier today. I (manually) cut a release of v3.4.0 (because Travis wasn't working, and not all the commits followed semantic-release).

I've also added those 3 as maintainers. There is some followup to do post merging that, but I can make issues for tracking them.

While there isn't a plan per se, there is work being done. Is that enough to close this? Perhaps.
